larrecodnn/ImagePatternAlgs/Tensorflow/Modules/EvaluateROIEff_module.cc

```diff
@@ -47,104 +56,433 @@ class nnet::EvaluateROIEff : public art::EDAnalyzer {
+
+        // check if signal/signals in this ROI
+        for (size_t s=0; s<signal_starttick.size(); s++) {
+          if (isSignalInROI(signal_starttick[s], signal_endtick[s], signal_max_tdctick[s], roiFirstBinTick, roiLastBinTick)) {
+            IsSigROI = true;
+            break;
+          }
+        } // loop over s
+      } // loop simChannels
+      
+      h_roi[view]->Fill(0); // total roi
+      if (IsSigROI) {
+        roi_sig[view] += 1. ;
+        fCount_Roi_sig[view] += 1;
+        h_roi[view]->Fill(1); // sig roi
+        h1_roi_max_sim[view]->Fill(maxadc_sig);
       }
+    } // loop ranges of signalROI
+  } // loop wires
+  
+  // purity of each plane for each event
+  for (int i=0; i<3; i++) {
+    if (roi_total[i]) {
+      double purity = roi_sig[i]/roi_total[i];
+      if (purity==1) purity = 1. -1.e-6;
+      h_purity[i]->Fill(purity);
     }
   }
+  
+  // combined purity for each event
+  if (roi_total[0]+roi_total[1]+roi_total[2]) {
+    double purity = (roi_sig[0]+roi_sig[1]+roi_sig[2])/(roi_total[0]+roi_total[1]+roi_total[2]);
+    if (purity==1) purity = 1. -1.e-6;
+    h_purity_all->Fill(purity);
+  }
 }
 
-void
-nnet::EvaluateROIEff::beginJob()
-{
+void nnet::EvaluateROIEff::beginJob(){
 
   art::ServiceHandle<art::TFileService const> tfs;
 
-  for (int i = 0; i < 3; ++i) {
-    h_energy[i] =
-      tfs->make<TH1D>(Form("h_energy_%d", i), Form("Plane %d; Energy (MeV);", i), 100, 0, 1);
-    h_energy_roi[i] =
-      tfs->make<TH1D>(Form("h_energy_roi_%d", i), Form("Plane %d; Energy (MeV);", i), 100, 0, 1);
+  for (int i = 0; i<3; ++i){
+    h_energy[i] = tfs->make<TH1D>(Form("h_energy_%d",i), Form("Plane %d; Energy (MeV);",i), 100,0,1);
+    h_energy_roi[i] = tfs->make<TH1D>(Form("h_energy_roi_%d",i), Form("Plane %d; Energy (MeV);",i), 100,0,1);
+
+    h_purity[i] = tfs->make<TH1D>(Form("h_purity_%d",i), Form("Plane %d; Purity;",i), 20,0,1);
+    h_roi[i] = tfs->make<TH1D>(Form("h_roi_%d",i), Form("Plane %d; Purity;",i), 5,0,5); // index 0: total rois; index 1: total sig rois
+    
+    h1_roi_max[i] = tfs->make<TH1D>(Form("h1_roi_max_%d",i), Form("Plane %d; Max adc;", i), 50,0,50);
+    h1_roi_max_sim[i] = tfs->make<TH1D>(Form("h1_roi_max_sim_%d",i), Form("Plane %d; Max adc;", i), 50,0,50);
+
+    h1_tickdiff_max[i] = tfs->make<TH1D>(Form("h1_tickdiff_max_%d",i), Form("Plane %d; tick diff (maxE - max pulse);", i), 100,-50,50);
   }
-}
 
-void
-nnet::EvaluateROIEff::endJob()
-{
+  h_purity_all = tfs->make<TH1D>("h_purity_all", "All Planes; Purity;", 20,0,1);
+}
 
+void nnet::EvaluateROIEff::endJob(){
   art::ServiceHandle<art::TFileService const> tfs;
 
-  for (int i = 0; i < 3; ++i) {
-    std::cout << i << " " << h_energy[i]->GetNbinsX() << " " << h_energy_roi[i]->GetNbinsX()
-              << std::endl;
-    if (TEfficiency::CheckConsistency(*(h_energy[i]), *(h_energy_roi[i]))) {
+  for (int i = 0; i<3; ++i){
+    std::cout<<i<<" "<<h_energy[i]->GetNbinsX()<<" "<<h_energy_roi[i]->GetNbinsX()<<std::endl;
```

Comment:
If printing out such information is necessary, please use `mf::LogInfo`, which allows a user to redirect or disable the logging of messages.